api/services/registration_service.py

```diff
@@ -52,7 +52,10 @@ def handle_registration(self, jsonData):
                         )
                     )
                     if user_program:
-                        models.UserProgram.query.update(user_program, user_program_data)
+                        if self.has_default_program_selection is False:
```

Comment:
In case, the user selects time but not program then the time will also not get updated. I think this check should be used for updating the value of `user_program_data["program_id"]`

Reply:
To update the time_slot user have to select the user_program. But got your point this check is only used for updating the value of `user_program_data["program_id"]`